agent/agent.go

```diff
@@ -462,35 +462,18 @@ func (a *Agent) setupTask(agentCtx, setupCtx context.Context, tcInput *taskConte
 	var taskConfig *internal.TaskConfig
 	taskConfig, err = a.makeTaskConfig(setupCtx, tc)
 	if err != nil {
-		setupFailed = true
-		err = errors.Wrap(err, "making task config")
-		grip.Error(err)
-		grip.Infof("Task complete: '%s'.", tc.task.ID)
-		tc.logger = client.NewSingleChannelLogHarness("agent.error", a.defaultLogger)
-		shouldExit, err := a.handleTaskResponse(setupCtx, tc, evergreen.TaskSystemFailed, err.Error())
-		return tc, setupFailed, shouldExit, err
+		return a.handleSetupError(errors.Wrap(err, "making task config"), tc, setupCtx, true)
 	}
 	tc.setTaskConfig(taskConfig)
 	if err = a.startLogging(agentCtx, tc); err != nil {
-		setupFailed = true
-		err = errors.Wrap(err, "setting up logger producer")
-		grip.Error(err)
-		grip.Infof("Task complete: '%s'.", tc.task.ID)
-		tc.logger = client.NewSingleChannelLogHarness("agent.error", a.defaultLogger)
-		shouldExit, err := a.handleTaskResponse(setupCtx, tc, evergreen.TaskSystemFailed, err.Error())
-		return tc, setupFailed, shouldExit, err
+		return a.handleSetupError(errors.Wrap(err, "setting up logger producer"), tc, setupCtx, true)
 	}
 
 	if !tc.ranSetupGroup {
 		tc.taskDirectory, err = a.createTaskDirectory(tc)
 		if err != nil {
-			setupFailed = true
-			err = errors.Wrap(err, "creating task directory")
-			grip.Error(err)
-			grip.Infof("Task complete: '%s'.", tc.task.ID)
 			tc.logger.Execution().Error(errors.Wrap(err, "creating task directory"))
```

Comment:
I don't think we actually need this log because the error is already passed to `handleSetupError`, which could just log the error to the `tc.logger` in there (this would work for the earlier cases as well, where the logger is not set up and we use the fallback one).

Reply:
but then we'd start logging errors that we weren't logging before like "making task config" and "setting up logger producer"